pkg/source/ttnv3/config.go

```diff
@@ -0,0 +1,128 @@
+package ttnv3
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/TheThingsNetwork/go-utils/handlers/cli"
+	ttnlog "github.com/TheThingsNetwork/go-utils/log"
+	ttnapex "github.com/TheThingsNetwork/go-utils/log/apex"
+	apex "github.com/apex/log"
+	"github.com/spf13/pflag"
+
+	"go.thethings.network/lorawan-stack-migrate/pkg/source/ttnv3/api"
+)
+
+var logger *apex.Logger
+
+type config struct {
+	appAccessKey string
+	appID        string
+	authToken    string
+
+	identityServerGRPCAddress    string
+	joinServerGRPCAddress        string
+	applicationServerGRPCAddress string
+	networkServerGRPCAddress     string
+
+	dryRun bool
+}
+
+func flagSet() *pflag.FlagSet {
+	flags := &pflag.FlagSet{}
+	flags.String(flagWithPrefix("app-id"), os.Getenv("TTNV3_APP_ID"), "TTS Application ID")
+	flags.String(flagWithPrefix("app-access-key"), os.Getenv("TTNV3_APP_ACCESS_KEY"), "TTS Application Access Key (with 'devices' permissions)")
+	flags.String(flagWithPrefix("ca-file"), os.Getenv("TTNV3_CA_FILE"), "TTS Path to a CA file (optional)")
+	flags.String(flagWithPrefix("identity-server-grpc-address"), os.Getenv("TTNV3_IDENTITY_SERVER_GRPC_ADDRESS"), "TTS Identity Server GRPC Address")
+	flags.String(flagWithPrefix("join-server-grpc-address"), os.Getenv("TTNV3_JOIN_SERVER_GRPC_ADDRESS"), "TTS Join Server GRPC Address")
+	flags.String(flagWithPrefix("application-server-grpc-address"), os.Getenv("TTNV3_APPLICATION_SERVER_GRPC_ADDRESS"), "TTS Application Server GRPC Address")
+	flags.String(flagWithPrefix("network-server-grpc-address"), os.Getenv("TTNV3_NETWORK_SERVER_GRPC_ADDRESS"), "TTS Network Server GRPC Address")
+	return flags
+}
+
+func getConfig(flags *pflag.FlagSet) (*config, error) {
+	stringFlag := func(f string) string {
+		s, _ := flags.GetString(f)
+		return s
+	}
+	boolFlag := func(f string) bool {
+		b, _ := flags.GetBool(f)
+		return b
+	}
+
+	appID := stringFlag(flagWithPrefix("app-id"))
+	if appID == "" {
+		return nil, errNoAppID
+	}
+
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{}
+	if caPath := stringFlag(flagWithPrefix("ca-file")); caPath == "" {
```

Comment:
Secure/insecure should be controllable by a flag (`--insecure`) and secure should be default. The CA field should be appended if provided, but can be missing.

Our production deployments use certificates that are signed by common CAs.